src/telemetry.rs

```diff
@@ -20,7 +20,9 @@ impl Telemetry {
 
         let (posthog, sentry_guard) = if config.enabled {
             let posthog = if let Some(key) = posthog_key {
-                Some(client(key.as_str()))
+                Some(client(PostHogClientOptions::new(key.as_str(), Some(&config.instance_id), sentry_dsn.is_none(), Some(|panic_exception| {
```

Comment:
Nit: probably it makes sense to set up a basic CI with `rustfmt` / `clippy`.

Reply:
Absolutely, just didn't want to mix it with this PR, but will do

Reply:
Added rustfmt + clippy